docs/csharp/programming-guide/classes-and-structs/static-constructors.md

```diff
@@ -11,27 +11,46 @@ ms.assetid: 151ec95e-3c4d-4ed7-885d-95b7a3be2e7d
 A static constructor is used to initialize any [static](../../../csharp/language-reference/keywords/static.md) data, or to perform a particular action that needs to be performed once only. It is called automatically before the first instance is created or any static members are referenced.  
   
  [!code-csharp[csProgGuideObjects#14](~/samples/snippets/csharp/VS_Snippets_VBCSharp/csProgGuideObjects/CS/Objects.cs#14)]  
-  
- Static constructors have the following properties:  
+ 
+## Remarks
+Static constructors have the following properties:  
   
 - A static constructor does not take access modifiers or have parameters.  
+
+- A class or struct can only have one static constructor.
+
+- Static constructors cannot be inherited or overloaded.
+
+- A static constructor cannot be called directly and is only meant to be called by the common language runtime (CLR). It is invoked automatically.
+
+- The user has no control on when the static constructor is executed in the program.
   
-- A static constructor is called automatically to initialize the [class](../../../csharp/language-reference/keywords/class.md) before the first instance is created or any static members are referenced. Note that a type's static constructor is called when a static method assigned to an event or a delegate is invoked and not when it is assigned.
-  
-- A static constructor cannot be called directly.  
-  
-- The user has no control on when the static constructor is executed in the program.  
+- A static constructor is called automatically to initialize the [class](../../../csharp/language-reference/keywords/class.md) before the first instance is created or any static members are referenced. A static constructor will run before an instance constructor. Note that a type's static constructor is called when a static method assigned to an event or a delegate is invoked and not when it is assigned. If static field variable initializers are present in the class of the static constructor, they will be executed in the textual order in which they appear in the class declaration immediately prior to the execution of the static constructor.
+
+- If you don't provide a static constructor to initialize static fields, the C# compiler will supply a default implicit static constructor that initializes static fields to their default value as listed in the [Default Values Table](../../../csharp/language-reference/keywords/default-values-table.md). 
```

Comment:
I'd like to rephrase this. The end result (initializing values to their default) is correct, but the compiler is not required to supply a static constructor. See the [last paragraph on default values in the C# language spec](https://docs.microsoft.com/dotnet/csharp/language-reference/language-specification/variables#default-values).

```suggestion
- If you don't provide a static constructor to initialize static fields, all static fields are initialized to their default value as listed in the [Default Values Table](../../../csharp/language-reference/keywords/default-values-table.md). 
```

Reply:
Appreciate the input.

That line was incorporated from [Constructors (C# Programming Guide)](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/constructors#static-constructors) :
>...If you don't provide a static constructor to initialize static fields, the C# compiler will supply a default static constructor that initializes static fields to their default value as listed in the Default Values Table.

Also, similar phrasing is used in [Static constructors and the TypeInitializationException](https://docs.microsoft.com/en-us/dotnet/api/system.typeinitializationexception?view=netframework-4.8#static-constructors-and-the-typeinitializationexception):
> ...Static constructors can be explicitly defined by a developer. If a static constructor is not explicitly defined, compilers automatically create one to initialize any static (in C#) or Shared (in Visual Basic) members of the type.

Should the text on these other pages be revised?

Reply:
😢 Yes, those pages should be updated as well. We're moving toward ensuring that the articles in these sections use language that's consistent with the spec (although often more informal). These should agree with the spec, and not imply different behavior.

Do you want to make those additional changes, or should I create an issue to update those docs and reference this conversation?

Reply:
I'm not sure how to best rephrase the text, so it'd be better if you were to file the issue.

Also, having looked at the IL output on [SharpLab](https://sharplab.io), I'm not sure the phrasing is incorrect.  When a static field is initialized at declaration, a `.cctor` is generated within which the static fields are initialized.

```csharp
//no .cctor
public class A{
    public static int x;
}
```
```csharp
//has .cctor 
//initializiation takes place in .cctor
public class B{
    public static int x=1;
}
```
```csharp
//has .cctor 
//initializiation takes place in .cctor
//x initialized before  y
public class C{
    public static int x=1;
    static C(){int y=2;}
}
```